### ISSUE TITLE
support FilScore

### ISSUE THREAD

**Issue by collinbarrett** [collinbarrett/FilterLists#371]
FilScore (codename, not finalized) will be a subjective rating for list quality. While it certainly would not be definitive, it would primarily provide for a better way to sort the main grid of lists by default than alphabetical.

Merging all of the individual issues into this one master issue in preparation for its initial development.

Scoring Factors:

- [ ] HTTPS Support (#335)
- [ ] Dead Weight (#337)

**Comment by DandelionSprout** [comment]
I am in favour of adding a user rating system, having been inspired by the system that is used on https://userstyles.org, as it'd provide an invaluable way for interested list maintainers to get any serious feedback on their lists, which is very difficult to get otherwise through today's methods.

However, I currently envision it in a way that'd go slightly against FilterLists.com's aim of ensuring full privacy:

• Allowing text comments would've been a near-must. If not, then the maintainer would've been left in the dark as to why his/her list got 2 stars from someone.
• Tying some sort of aliases to the text comments would've also probably been needed, but is not a matter of life or death.
• I honestly don't think that a pure numerical score would've been a good idea, as it'd fail to account properly for technical issues on the user-end, or for potentially controversial lists (*Acceptable Ads* come to mind) that'd possibly attract ★☆☆☆☆ reviews in droves. I imagine a more fluid scale, whose score options would be á la "This list is awesome", "This list is good", "I couldn't load the list", "I didn't like this list", "The entries didn't do anything", etc.

**Comment by DandelionSprout** [comment]
It's probably pretty late of me to mention this, but I've begun to realise that there's a bit of a problem regarding `Software Inclusion/Default`.

Basically, when it comes to non-regional lists, Fanboy's and AdGuard's lists seem to be grandfathered in uBO and Nano, to the detriment of virtually every other list maker out there. Of the 13 lists in uBO's *Ads*, *Privacy*, and *Annoyances* sections, 5 are from Fanboy, 5 are from Adguard, 3 are from EasyList, while *no one else are represented*.

Sure, this may just be me being salty about how *Fanboy's Cookie List* and *Fanboy's Annoyances List* is included in uBO, despite how he's from New Zealand and can't even see half the cookie things that his lists are supposed to block.

So in essence, I wouldn't have wanted to place a whole lot of weight on `Software Inclusion/Default`, as it currently stands.

**Comment by collinbarrett** [comment]
Thanks, @DandelionSprout . Added your concern to the OP. At this point, adding this "FilScore" (not a fan of that name, really) seems like a dream. Life has just been so busy that I haven't had time to give much new development to this project, as much as I want to. Hoping that changes at some point. But, I totally see your point. Good feedback.

**Comment by jawz101** [comment]
I agree with DandelionSprout.  I wouldn't even have it as a data point to score.  The same with user votes as well, however.

I'd just score on this:

- Dead Weight Inclusion
- Uniqueness/Furthest Upstream difficult to tell :/
- Update Frequency
- Generic CDN Domain Inclusion/**Subdomain enumeration** [example](https://github.com/lightswitch05/hosts/issues/76) run list against a list of regex'd subdomain/CDN's [example list](https://github.com/lightswitch05/hosts/blob/master/ads-and-tracking-whitelist.txt)
- and maybe give it a Subjectivity weight by running it against [Alexa's](https://support.alexa.com/hc/en-us/articles/200449834-Does-Alexa-have-a-list-of-its-top-ranked-websites-) top X sites, [Majestic](https://blog.majestic.com/development/majestic-million-csv-daily/) or [OpenDNS](http://s3-us-west-1.amazonaws.com/umbrella-static/index.html).  I can see blocking tracker.facebook.com or analytic.google.com but specifically blocking Facebook is pretty opinionated and I'd like to think a blocklist is for us to consume without ruining the end user experience.

Number of hosts 30,000
30% dead weight
updated in last 60 days
5% expected breakage (includes popular web page domains or enumerations)

I'd give something like Easylist a C [personally](https://github.com/easylist/easylist/issues/2374).  For such a popular list, having 30% of it as dead weight just bugs me.
